### PR TITLE
Refactor Profile Page UI

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
Room Details Page

- - -

https://github.com/acterglobal/a3/assets/51526480/265dceed-315e-4be7-9eb1-c795a145c8ac

### CODE REVIEW

**Comment by gnunicorn** [comment]
@kumarpalsinh25 we opted for not showing the members right away because that is a potentially large list that takes quite a bit to load. If that is offloaded into its own consumer/widget that doesn't block the other parts of the UI we can consider this design, otherwise we have to stick with it being a menu item ...

File: app/lib/features/chat/pages/room_profile_page.dart [hunk 1]
```diff
@@ -70,282 +74,314 @@ class RoomProfilePage extends ConsumerWidget {
...
+                      );
+              },
+            );
+          },
+          error: (e, st) => Text('Error loading tile due to $e'),
+          loading: () => const ActionItemSkeleton(
+            iconData: Atlas.user_plus_thin,
+            actionName: 'Invite',
+          ),
+        ),
```
Comment (gnunicorn) [inline app/lib/features/chat/pages/room_profile_page.dart:202]: arguably showing the invite if you can't invite is a bit confusing. maybe we could show a disabled button with a tooltip stating the user can't invite to this room instead?

File: app/lib/features/chat/pages/room_profile_page.dart [hunk 2]
```diff
@@ -70,282 +74,314 @@ class RoomProfilePage extends ConsumerWidget {
...
+          context: context,
+          iconData: Icons.ios_share,
+          actionName: 'Share',
+          onTap: () async {
+            final roomLink =
+                await (await ref.read(chatProvider(widget.roomId).future))
+                    .permalink();
+            Share.share(
+              roomLink,
+              subject: 'Room ID',
```
Comment (gnunicorn) [inline app/lib/features/chat/pages/room_profile_page.dart:215]:
```suggestion
              subject: 'Link to Chat',
```

**Review by gnunicorn — APPROVED** [review]
LGTM. Just some minor stuff, totally up for you if you want to do them before merging.

File: app/lib/features/chat/pages/room_profile_page.dart (hunk 1 above)
Comment (kumarpalsinh25) [reply app/lib/features/chat/pages/room_profile_page.dart:202]: Let me change colour of invite action button if user don't have permission.

Comment (kumarpalsinh25) [reply app/lib/features/chat/pages/room_profile_page.dart]: Done :)
Screenshot 2024-03-08 at 10 03 51 PM

**Pull request merged**